tests/cross/src/test/scala/tests/pc/SelectionRangeSuite.scala

```diff
@@ -179,4 +179,40 @@ class SelectionRangeSuite extends BaseSelectionRangeSuite {
     )
   )
 
+  check(
+    "def - braceless",
+    """object Main extends App :
+      |  def foo(hi: Int, b@@: Int, c:Int) = ???  """.stripMargin,
+    List(),
+    Map(
+      "3" -> List(
+        """object Main extends App :
+          |  def foo(hi: Int, >>region>>b: Int<<region<<, c:Int) = ??? """.stripMargin,
+        """object Main extends App :
+          |  def foo(>>region>>hi: Int, b: Int, c:Int<<region<<) = ??? """.stripMargin,
+        """object Main extends App :
+          |  >>region>>def foo(hi: Int, b: Int, c:Int) = ???<<region<< """.stripMargin,
+        """object Main extends >>region>>App :
+          |  def foo(hi: Int, b: Int, c:Int) = ???<<region<<""".stripMargin,
+        """>>region>>object Main extends App :
+          |  def foo(hi: Int, b: Int, c:Int) = ???<<region<<""".stripMargin
+      )
+    )
+  )
+
+  check(
+    "def - braced",
+    """object Main extends App { def foo(hi: Int, b@@: Int, c:Int) = ??? } """.stripMargin,
+    List(),
+    Map(
+      "3" -> List(
+        """object Main extends App { def foo(hi: Int, >>region>>b: Int<<region<<, c:Int) = ??? }""".stripMargin,
+        """object Main extends App { def foo(>>region>>hi: Int, b: Int, c:Int<<region<<) = ??? }""".stripMargin,
+        """object Main extends App { >>region>>def foo(hi: Int, b: Int, c:Int) = ???<<region<< }""".stripMargin
+        /* These fail. A perfect implemenation would succeed (I believe). The effort / fix value ratio means we leave it as is for now */
```

Comment:
You can create a follow up feature request

Reply:
https://github.com/scalameta/metals/issues/6569